dozer-ingestion/javascript/src/js_extension/ts_module_loader.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+//! This example shows how to use swc to transpile TypeScript and JSX/TSX
+//! modules.
+//!
+//! It will only transpile, not typecheck (like Deno's `--no-check` flag).
+
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::pin::Pin;
+use std::rc::Rc;
+
+use deno_runtime::deno_core::{self, anyhow, futures};
+
+use anyhow::anyhow;
+use anyhow::bail;
+use anyhow::Error;
+use deno_ast::MediaType;
+use deno_ast::ParseParams;
+use deno_ast::SourceTextInfo;
+use deno_core::error::AnyError;
+use deno_core::resolve_import;
+use deno_core::ModuleLoader;
+use deno_core::ModuleSource;
+use deno_core::ModuleSourceFuture;
+use deno_core::ModuleSpecifier;
+use deno_core::ModuleType;
+use deno_core::ResolutionKind;
+use deno_core::SourceMapGetter;
+use futures::FutureExt;
+
+#[derive(Clone)]
+struct SourceMapStore(Rc<RefCell<HashMap<String, Vec<u8>>>>);
+
+impl SourceMapGetter for SourceMapStore {
+    fn get_source_map(&self, specifier: &str) -> Option<Vec<u8>> {
+        self.0.borrow().get(specifier).cloned()
+    }
+
+    fn get_source_line(&self, _file_name: &str, _line_number: usize) -> Option<String> {
+        None
+    }
+}
+
+pub struct TypescriptModuleLoader {
```

Comment:
What is the difference with `deno_core::FsModuleLoader`?

Reply:
`TypescriptModuleLoader` calls `swc` to support modern syntax and TypeScript.